### ISSUE TITLE
Repository sync-to: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by cristihcd** [kopia/kopia#1303]
Hello,

After having problems with high memory usage using kopia server (https://github.com/kopia/kopia/issues/1233#issuecomment-912345342) I tried to synchronize a local repo to a remote kopia server.

I connected to the kopia server using a specific config file then ran: 
`kopia repository sync-to from-config --file=./config/kopia/server/repository.config`

I get the following error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf2a2aa]

goroutine 1 [running]:
github.com/kopia/kopia/cli.(*storageFromConfigFlags).connectToStorageFromConfigFile(0x0, {0x17ac058, 0xc00012a000})
        /home/runner/work/kopia/kopia/cli/command_repository_connect_from_config.go:50 +0x6a
github.com/kopia/kopia/cli.(*storageFromConfigFlags).connect(0x0, {0x17ac058, 0xc00012a000}, 0x0)
        /home/runner/work/kopia/kopia/cli/command_repository_connect_from_config.go:33 +0xb4
github.com/kopia/kopia/cli.(*commandRepositorySyncTo).setup.func1(0x0)
        /home/runner/work/kopia/kopia/cli/command_repository_sync.go:61 +0x88
github.com/alecthomas/kingpin.(*actionMixin).applyActions(0x1, 0xc0001e0b60)
        /home/runner/go/pkg/mod/github.com/alecthomas/kingpin@v0.0.0-20200323085623-b6657d9477a6/actions.go:28 +0x5c
github.com/alecthomas/kingpin.(*Application).applyActions(0xc0000ec500, 0xc0001a4240)
        /home/runner/go/pkg/mod/github.com/alecthomas/kingpin@v0.0.0-20200323085623-b6657d9477a6/app.go:560 +0xa2
github.com/alecthomas/kingpin.(*Application).execute(0xc0000ec500, 0xc0001a4240, {0xc000119180, 0x3, 0x4})
        /home/runner/go/pkg/mod/github.com/alecthomas/kingpin@v0.0.0-20200323085623-b6657d9477a6/app.go:398 +0x66
github.com/alecthomas/kingpin.(*Application).Parse(0xc0000ec500, {0xc000114130, 0x53, 0x177abf0})
        /home/runner/go/pkg/mod/github.com/alecthomas/kingpin@v0.0.0-20200323085623-b6657d9477a6/app.go:230 +0x165
main.main()
        /home/runner/work/kopia/kopia/main.go:83 +0x30c
```

Versions:
kopia server: 0.9.0-rc1, Windows
kopia client: 20210919.0.185038, Debian 11

Thanks for looking into this.

**Comment by jkowalski** [comment]
This operation is not supported and the error message should be improved. Sync-to only works with direct repositories not server-based repositories. 

Can you describe your scenario in more detail, what exactly are you're trying to achieve?

**Comment by cristihcd** [comment]
I am trying to backup a directory with many small files (). As described in https://github.com/kopia/kopia/issues/1233#issuecomment-912345342, I ran into memory usage issues using kopia client -> kopia repository server.

After that, I ran a backup on a local filesystem repository, then tried to copy it over to the repository server. Now I realize I should have used [snapshot copy-history](https://kopia.io/docs/reference/command-line/common/snapshot-copy-history/) instead.

However, the original issue remains, that I cannot backup directly to the repository server, due to very high memory usage.
